src/lib.rs

```diff
@@ -175,17 +175,20 @@ impl AutoCfg {
         let rustflags = if target != env::var_os("HOST")
             || dir_contains_target(&target, &dir, env::var_os("CARGO_TARGET_DIR"))
         {
-            env::var("RUSTFLAGS").ok().map(|rustflags| {
-                // This is meant to match how cargo handles the RUSTFLAG environment
-                // variable.
-                // See https://github.com/rust-lang/cargo/blob/69aea5b6f69add7c51cca939a79644080c0b0ba0/src/cargo/core/compiler/build_context/target_info.rs#L434-L441
-                rustflags
-                    .split(' ')
-                    .map(str::trim)
-                    .filter(|s| !s.is_empty())
-                    .map(str::to_string)
-                    .collect::<Vec<String>>()
-            })
+            env::var("CARGO_ENCODED_RUSTFLAGS")
+                .or_else(|_| env::var("RUSTFLAGS"))
+                .ok()
+                .map(|rustflags| {
+                    // This is meant to match how cargo handles the RUSTFLAG environment
+                    // variable.
+                    // See https://github.com/rust-lang/cargo/blob/69aea5b6f69add7c51cca939a79644080c0b0ba0/src/cargo/core/compiler/build_context/target_info.rs#L434-L441
+                    rustflags
+                        .split(' ')
```

Comment:
You have to use `'\x1f'` for `CARGO_ENCODED_RUSTFLAGS`.

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts

> extra flags that Cargo invokes `rustc` with, separated by a `0x1f` character (ASCII Unit Separator).

Reply:
Good catch! I've updated my PR accordingly.